subsys/bluetooth/host/l2cap_br.c

```diff
@@ -201,7 +201,7 @@ static bool l2cap_br_chan_add(struct bt_conn *conn, struct bt_l2cap_chan *chan,
 		return false;
 	}
 
-	k_delayed_work_init(&chan->rtx_work, l2cap_br_rtx_timeout);
+	k_work_init_delayable(&chan->rtx_work, l2cap_br_rtx_timeout);
```

Comment:
This appears to risk corrupting a work structure while the kernel owns it, given so many unchecked `k_work_cancel_delayable()` calls without comments explaining why they're safe.

Reply:
Yes, that appears to be true for the dynamic channels, where the channels can be deleted while the connection exists.
I have added cancel_sync to the destroy handler which should handle this.